File: 0367-valid-perfect-square/0367-valid-perfect-square.cpp
class Solution {
public:
    double Sqrt(int n) {
        if (n < 2)
            return double(n);

        double sqrt = n / 2;
        double temp = 0;

        while (sqrt != temp) {
            temp = sqrt;
            sqrt = (n / temp + temp) / 2;
        }

        return sqrt;
    }

    bool isPerfectSquare(int num) {
        double a = Sqrt(num);
        double b = ceil(Sqrt(num));
        if (a == b)
            return true;
        return false;
    }
};

File: 0367-valid-perfect-square/0367-valid-perfect-square.java
class Solution {
    public boolean isPerfectSquare(int num) {
        int i = 1; // Start with i = 1
        while (num > 0) {
            num -= i; // Subtract the current odd number from num
            i += 2;   // Increment i by 2 to get the next odd number
        }
        return num == 0; // If num becomes 0, it's a perfect square
    }
}
